docs/testplan/snappi/srv6_snake_test.md

```diff
@@ -0,0 +1,157 @@
+# SONiC Switch SRv6 Dataplane Performance Test -- Snake Stress Testing Setup
+
+- [Test Objective](#test-objective)
+- [Test Setup](#test-setup)
+- [Test Parameters](#test-parameters)
+- [Test Steps](#test-steps)
+- [Metrics to Collect](#metrics-to-collect)
+
+## Test Objective
+
+This test aims to assess the data-plane performance of the SRv6 forwarding function of a SONiC switch in an approximately full load condition.
+
+## Test Setup
+
+### Network Topology Setup
+
+- The test is designed to be run againts a single device.
+- The device under test(DUT) should have a small subset of its ports connected to the traffic generators while all the other ports connected to each other to form a snake topology.
+
+An example of the topology is shown in the following figure:
+
+![Snake Topology](./snake_topo.png)
+
+### Network Configuration
+
+The DUT should have SRv6 and route configurations as follows:
+- The DUT should be configured with a number of SRv6 SIDs each of which corresponds to an active switch port. If using fcbb:bbbb:: as the locator block, the SRv6 SIDs of a switch with Q ports can be configured to be fcbb:bbbb:1::/48 ~ fcbb:bbbb:hex(Q)::/48.
+- Every Traffic Generator should also be configured a number of SRv6 SIDs each of which corresponds to a link between the traffic generator and the DUT. If using fcbb:bbbb:: as the locator block, the SRv6 SIDs of a traffic generator with N ports can be as fcbb:bbbb:100::/48 ~ fcbb:bbbb:hex(N)00::/48.
+- The DUT should have a static route entry configured for each SRv6 SID that it has and the traffic generators have.
+
+### Traffic Generation Configuration
+
+The traffic generators should be configured to send traffic as follows:
+
+For a port of the traffic generator indexed by i (1 <= i <= N), the packet sent by the traffic generator should have SRv6 SID list in IPv6 header (and potentially Segment Routing Header) as follows:
+- fcbb:bbbb:i00:i:hex(N + i):hex(2N + i)...hex(MN + i):hex(N+i)00::, note: hex(N+i)00 refers to the SRv6 SID of the receiving traffic generator.
+
+To maximize the stress on the DUT, the i-th port of the other traffic generator should send packets to the DUT with SRv6 SID list as follows:
+
+- fcbb:bbbb:hex(N+i)00:hex(MN + i):hex((M-1)N + i):...:i:i00::, note: this is essentially the reverse of the SID list used by the other side.
+
+### Metrics Monitoring
```

Comment:
looks like the test steps, metrics monitoring can simply reference the capability test cases, and we can highlight only the differences here.

the capability test update PR is already merged, so we can use the link directly.